src/lib/Util/adapters/ibm/index.js

```diff
@@ -0,0 +1,232 @@
+        "offset": 9,
+        "length": 1,
+        "key": "ngomd9"
+      },
+      {
+        "start": 13.44,
+        "end": 13.86,
+        "confidence": 1,
+        "text": "day.",
+        "offset": 11,
+        "length": 4,
+        "key": "sgmfl4f"
+      }
+    ]
+  },
+  ...
+```
+ *
+ */
+import generateEntitiesRanges from '../generate-entities-ranges/index.js';
+
+const ibmToDraft = ibmJson => {
+  // helper function to normalise IBM words at line level
+  const normalizeTimeStampsToWords = timestamps => {
+    return timestamps.map(ibmWord => {
+      return {
+        text: ibmWord[0],
+        start: ibmWord[1],
+        end: ibmWord[2]
+      };
+    });
+  };
+
+  //
+  const normalizeIBMWordsList = ibmResults => {
+    const normalisedResults = [];
+    ibmResults.forEach(result => {
+      // nested array to keep paragraph segmentation same as IBM lines
+      normalisedResults.push(normalizeTimeStampsToWords(result.alternatives[0].timestamps));
+      // TODO: can be revisited - as separate PR by flattening the array like this
+      // normalisedResults = normalisedResults.concact(normalizeTimeStampsToWords(result.alternatives[0].timestamps));
+      // addSpeakersToWords function would need adjusting as would be dealing with a 1D array instead of 2D
+      // if edge case, like in example file, that there's one speaker recognised through all of speaker segemtnation info
+      // could break into paragraph when is over a minute? at end of IBM line?
+      // or punctuation, altho IBM does not seem to provide punctuation?
+    });
+
+    return normalisedResults;
+  };
+
+  // TODO: could be separate file
+  const findSpeakerSegmentForWord = (word, speakerSegments) => {
+    const tmpSegment = speakerSegments.find(seg => {
+      const segStart = seg.from;
+      const segEnd = seg.to;
+
+      return ((word.start === segStart) && (word.end === segEnd));
+    });
+    // if find doesn't find any matches it returns an undefined
+    if (tmpSegment === undefined) {
+      // covering edge case orphan word not belonging to any segments
+      // adding UKN speaker label
+      return 'UKN';
+    } else {
+      // find returns the first element that matches the criteria
+      return `S_${ tmpSegment.speaker }`;
+    }
+  };
+  // add speakers to words
+  const addSpeakersToWords = (ibmWords, ibmSpeakers) => {
+    return ibmWords.map(lines => {
+      return lines.map(word => {
+
+        word.speaker = findSpeakerSegmentForWord(word, ibmSpeakers);
+
+        return word;
+      });
+    });
+  };
+
+  const ibmNormalisedWordsToDraftJs = (ibmNormalisedWordsWithSpeakers) => {
+    const drafJsParagraphsResults = [];
+    ibmNormalisedWordsWithSpeakers.forEach((ibmParagraph) => {
+      const draftJsContentBlockParagraph = {
+        text: ibmParagraph.map((word) => {return word.text;}).join(' '),
+        type: 'paragraph',
+        data: {
+          // TODO: Assuming each paragraph in IBM line is the same
```

Comment:
does more work need to be done here?

Reply:
Don't think so, It just seems like the IBM data structure gives you word level speakers,but also gives you "lines" so assuming each word in a line has the same speaker.

Can also check with @seanmtracey if has time to review this later.

Reply:
removing the todo and expanding on the comment to give more context.